be/src/vec/functions/function_string.h

```diff
@@ -1940,22 +1940,20 @@ class FunctionMoneyFormat : public IFunction {
         ColumnPtr argument_column = block.get_by_position(arguments[0]).column;
 
         auto result_column = assert_cast<ColumnString*>(res_column.get());
-        auto data_column = assert_cast<const typename Impl::ColumnType*>(argument_column.get());
```

Comment:
I think the origin way is better? why change the logic, maybe `template MoneyFormatDecimalImpl ` is better